torch/csrc/autograd/variable.h

```diff
@@ -376,7 +377,7 @@ struct TORCH_API Variable::AutogradMeta : public c10::AutogradMetaInterface {
   }
 
   bool requires_grad() const override {
-    return requires_grad_ || grad_fn_;
+    return (requires_grad_ || grad_fn_) && GradMode::is_enabled();
```

Comment:
Can we make `is_enabled` an inline function to make it equivalent to checking a single boolean? I think that right now it's defined as a function, so this might require an extra call and `.requires_grad` should be blazing fast.